python/tests/unit/arcticdb/version_store/test_column_type_changes.py

```diff
@@ -122,4 +125,270 @@ def get_type_of_column():
 
     result_df = lib.read(sym).data
     expected_df = pd.concat([df_write, df_append_1, df_append_2])
-    assert_frame_equal(result_df, expected_df)
\ No newline at end of file
+    assert_frame_equal(result_df, expected_df)
+
+
+@pytest.mark.parametrize("int_type", [np.int64, np.uint64, np.int32, np.uint32])
+@pytest.mark.parametrize("float_type", float_types)
+@pytest.mark.parametrize("second_append_type", [np.int64, np.uint64, np.int32, np.uint32])
+@pytest.mark.parametrize("int_first", (True, False))
+def test_type_promotion_ints_and_floats_up_to_float64(lmdb_version_store_dynamic_schema, int_type, float_type, second_append_type, int_first):
+    # Given
+    lib = lmdb_version_store_dynamic_schema
+
+    if int_first:
+        original_data = pd.DataFrame({"a": np.array([1, 2, 3], int_type)}, index=[0, 1, 2])
+        first_append = pd.DataFrame({"a": np.array([4.0, 5.0, 6.0], float_type)}, index=[3, 4, 5])
+    else:
+        original_data = pd.DataFrame({"a": np.array([1, 2, 3], float_type)}, index=[0, 1, 2])
+        first_append = pd.DataFrame({"a": np.array([4.0, 5.0, 6.0], int_type)}, index=[3, 4, 5])
+
+    second_append = pd.DataFrame({"a": np.array([7, 8, 9], second_append_type)}, index=[5, 6, 7])
```

Comment:
What does the second_append achieve?

Reply:
I wanted to have some kind of check that we haven't messed up the type descriptors after the first append and the promotion, so subsequent promotions behave the same as the first promotion.